src/core/ext/transport/chttp2/transport/flow_control.h

```diff
@@ -309,7 +309,7 @@ class TransportFlowControl final : public TransportFlowControlBase {
     return action;
   }
 
-  const grpc_chttp2_transport* const t_;
```

Comment:
why?

Reply:
Iirc it's because `grpc_chttp2_stream_map_for_each` requires a non-const field. I can probably write a const version of it too though

Reply:
I think that would be preferable... just to limit the scope of potential mutations.

Reply:
Maybe we can track some kind of crude approximation and recheck iff that approximation says we'd overflow?

Reply:
Is this really that much though? Internally, I've seen 100 as the number being thrown around as the configuration for maximum number of concurrent streams?

If it is an issue, we could probably say something like - the initial window size can only grow as much as half the maximum flow control window and a similar restriction on the window delta.